### DISCUSSION TITLE
Change linkerd-proxy container name

### DISCUSSION THREAD

**Discussion by realhuseyn** [linkerd/linkerd2#10214] [Help]
Hello. I have a question. Can we change the default container name of the injected proxy container (linkerd-proxy). I didn't find any configuration related to it. My purpose is to reorder containers for my pods. I want that the first container of my pod is my app container, not linkerd-proxy container.

**Comment by mateiidavid** [comment]
@realhuseyn we don't support changing the name of the container. Although containers are started concurrently by the kubelet, linkerd-proxy has a [postStart hook](https://kubernetes.io/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/) to ensure it starts up before the application container. It needs to be declared first so that the hook blocks your application container from starting. This ensures the proxy is up before your application so traffic won't fail.

Even if you did manage to re-order them, that wouldn't achieve much on its own, containers would still start at the same time without the use of hooks. Why do you need them re-ordered?

**Reply by realhuseyn** [reply]
@mateiidavid thank you for your response. I want my application container as the first container. When I execute "kubectl logs -n namespace my_pod_name" I receive the logs of linkerd-proxy due to the first container. I must use "-c my_application_container_name" for this purpose. If the first container is my application container, I will receive the needed logs without "-c my_application_container_name".

**Reply by kleimkuhler** [reply]
@realhuseyn you can disable this with `--set proxy.await="false"`. You'll disable `linkerd-await` ensuring that the proxy container is running before the application container, but it will also inject the proxy after all other containers. You can see [here](https://github.com/linkerd/linkerd2/blob/309e8d12102b26997267ef720a0af9b76dd31ab5/charts/patch/templates/patch.json#L106-L110) in the code that when `proxy.await` is enabled, we inject the proxy container before all other containers.
